projects/igniteui-angular/src/lib/grids/hierarchical-grid/hierarchical-grid.component.ts

```diff
@@ -588,9 +592,31 @@ export class IgxHierarchicalGridComponent extends IgxHierarchicalGridBaseCompone
 
     private hg_verticalScrollHandler(event) {
         this.scrollTop = event.target.scrollTop;
+        this.hideOverlays();
+    }
+
+    public onContainerScrol() {
+        this.hideOverlays();
     }
 
     private hg_horizontalScrollHandler(event) {
         this.scrollLeft = event.target.scrollLeft;
+        this.hideOverlays();
+    }
+
+    private hideOverlays() {
+        const allOverlays = (this.overlayService as any)._overlayInfos;
```

Comment:
@MayaKirova This should be rewritten so that the grid keeps the ids of the opened overlays and just iterates over the collection to close them.